tests/operators/test_agnostic_load_file.py

```diff
@@ -466,47 +468,94 @@ def test_aql_gcs_file_to_postgres(self):
+
+@pytest.fixture
+def sample_dag():
+    yield DAG("test_dag", default_args={"owner": "airflow", "start_date": DEFAULT_DATE})
+    with create_session() as session:
+        session.query(DagRun).delete()
+        session.query(TI).delete()
+
+
+def create_and_run_task(dag, decorator_func, op_args, op_kwargs):
+    with dag:
+        function = decorator_func(*op_args, **op_kwargs)
+
+    _ = dag.create_dagrun(
+        run_id=DagRunType.MANUAL.value,
+        start_date=timezone.utcnow(),
+        data_interval=[DEFAULT_DATE, DEFAULT_DATE],
+        execution_date=DEFAULT_DATE,
+        state=State.RUNNING,
+    )
+    function.operator.run(start_date=DEFAULT_DATE, end_date=DEFAULT_DATE)
+    return function
+
+
+@pytest.fixture
+def hook(request):
+    if request.param == "snowflake":
         hook = SnowflakeHook(
             snowflake_conn_id="snowflake_conn",
             schema=os.getenv("SNOWFLAKE_SCHEMA"),
             database=os.getenv("SNOWFLAKE_DATABASE"),
             warehouse=os.getenv("SNOWFLAKE_WAREHOUSE"),
         )
-
-        # Drop target table
         hook.run(
-            f"DROP TABLE IF EXISTS {os.getenv('SNOWFLAKE_SCHEMA')}.{self.SNOWFLAKE_OUTPUT_TABLE_NAME}"
+            f"DROP TABLE IF EXISTS {os.getenv('SNOWFLAKE_SCHEMA')}.{OUTPUT_TABLE_NAME}"
         )
-        self.create_and_run_task(
-            load_file,
-            (),
-            {
-                "path": str(self.cwd) + "/../data/homes.csv",
-                "file_conn_id": "",
-                "output_table": Table(
-                    table_name=self.SNOWFLAKE_OUTPUT_TABLE_NAME,
-                    database=os.getenv("SNOWFLAKE_DATABASE"),
-                    schema=os.getenv("SNOWFLAKE_SCHEMA"),
-                    warehouse=os.getenv("SNOWFLAKE_WAREHOUSE"),
-                    conn_id="snowflake_conn",
-                ),
-            },
-        )
-
-        # Read table from db
-        df = hook.get_pandas_df(
-            f"SELECT * FROM {os.getenv('SNOWFLAKE_SCHEMA')}.{self.SNOWFLAKE_OUTPUT_TABLE_NAME}"
+        yield hook
+        hook.run(
+            f"DROP TABLE IF EXISTS {os.getenv('SNOWFLAKE_SCHEMA')}.{OUTPUT_TABLE_NAME}"
         )
-
-        assert df.iloc[0].to_dict() == {
-            "SELL": 142.0,
-            "LIST": 160.0,
-            "LIVING": 28.0,
-            "ROOMS": 10.0,
-            "BEDS": 5.0,
-            "BATHS": 3.0,
-            "AGE": 60.0,
-            "ACRES": 0.28,
-            "TAXES": 3167.0,
+    elif request.param == "postgres":
+        POSTGRES_TABLE_NAME = "test_dag_load_file_homes_csv_2"
+        hook = TempPostgresHook(postgres_conn_id="postgres_conn", schema="pagila")
+        hook.run(f"DROP TABLE IF EXISTS tmp_astro.{OUTPUT_TABLE_NAME}")
+        yield hook
+        hook.run(f"DROP TABLE IF EXISTS tmp_astro.{OUTPUT_TABLE_NAME}")
+    else:
+        raise ValueError(f"Unsupported database {request.param}")
+
+
+@pytest.mark.parametrize("hook", ["snowflake", "postgres"], indirect=True)
```

Comment:
Would it make sense to parametrize on "output table" param instead of hooks? that way when we introduce multiple DB support we just need to add to the list of output table?